scripts/fetch_olean_cache.sh

```diff
@@ -15,7 +15,18 @@ done
 # exit if there were no successful requests
 [ "$new_git_sha" != "" ] || exit 0
 
-curl "$archive_url$new_git_sha.tar.xz" | tar xJ src
+# A list of directories containing .olean files. We are being conservative to
+# avoid traversing irrelevant directories and affecting directories we do not
+# want changed (e.g. $root/.git).
+dirs="src"
+
+# if there are errors extracting the olean cache, delete all oleans and continue
+{ curl "$archive_url$new_git_sha.tar.xz" | tar xJ src; } || {
```

Comment:
```suggestion
(curl "$archive_url$new_git_sha.tar.xz" | tar xJ src) || {
```
I'm not sure if you're using the curly braces for performance or semantic reasons.

Reply:
Neither, I was confused about how variables and subshells worked.